pkg/distributor/distributor_test.go

```diff
@@ -3786,30 +3820,25 @@ func TestDistributor_Push_RelabelDropWillExportMetricOfDroppedSamples(t *testing
 	})
 
 	// Push the series to the distributor
+	id := "user"
 	req := mockWriteRequest(inputSeries, 1, 1, false)
-	ctx := user.InjectOrgID(context.Background(), "userDistributorPushRelabelDropWillExportMetricOfDroppedSamples")
+	req.Timeseries[0].Exemplars = []cortexpb.Exemplar{
+		{Labels: cortexpb.FromLabelsToLabelAdapters(labels.FromStrings("test", "a")), Value: 1, TimestampMs: 0},
+		{Labels: cortexpb.FromLabelsToLabelAdapters(labels.FromStrings("test", "b")), Value: 1, TimestampMs: 0},
+	}
+	ctx := user.InjectOrgID(context.Background(), id)
 	_, err = ds[0].Push(ctx, req)
 	require.NoError(t, err)
 
-	// Since each test pushes only 1 series, we do expect the ingester
-	// to have received exactly 1 series
 	for i := range ingesters {
 		timeseries := ingesters[i].series()
 		assert.Equal(t, 1, len(timeseries))
 	}
 
-	metrics := []string{"cortex_distributor_received_samples_total", "cortex_discarded_samples_total"}
-
-	expectedMetrics := `
-		# HELP cortex_discarded_samples_total The total number of samples that were discarded.
-		# TYPE cortex_discarded_samples_total counter
-		cortex_discarded_samples_total{reason="relabel_configuration",user="userDistributorPushRelabelDropWillExportMetricOfDroppedSamples"} 1
-		# HELP cortex_distributor_received_samples_total The total number of received samples, excluding rejected and deduped samples.
-		# TYPE cortex_distributor_received_samples_total counter
-        cortex_distributor_received_samples_total{type="float",user="userDistributorPushRelabelDropWillExportMetricOfDroppedSamples"} 1
-        cortex_distributor_received_samples_total{type="histogram",user="userDistributorPushRelabelDropWillExportMetricOfDroppedSamples"} 0
-		`
-	require.NoError(t, testutil.GatherAndCompare(regs[0], strings.NewReader(expectedMetrics), metrics...))
+	require.Equal(t, testutil.ToFloat64(ds[0].validateMetrics.DiscardedSamples.WithLabelValues(validation.DroppedByRelabelConfiguration, id)), float64(1))
+	require.Equal(t, testutil.ToFloat64(ds[0].validateMetrics.DiscardedExemplars.WithLabelValues(validation.DroppedByRelabelConfiguration, id)), float64(2))
+	require.Equal(t, testutil.ToFloat64(ds[0].receivedSamples.WithLabelValues(id, "float")), float64(1))
+	require.Equal(t, testutil.ToFloat64(ds[0].receivedSamples.WithLabelValues(id, "histogram")), float64(0))
```

Comment:
Should we test ingesting a native histogram and see it be discarded?

Reply:
Updated in latest commit